ext/nnstreamer/tensor_decoder/tensordec-directvideo.c

```diff
@@ -30,47 +30,131 @@
 #include <glib.h>
 #include <gst/video/video-format.h>
 #include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
 #include <nnstreamer_log.h>
 #include "tensordecutil.h"
 
 void init_dv (void) __attribute__ ((constructor));
 void fini_dv (void) __attribute__ ((destructor));
 
 #define DECODER_DV_VIDEO_CAPS_STR \
-    GST_VIDEO_CAPS_MAKE ("{ RGB, BGRx, GRAY8 }") \
+    GST_VIDEO_CAPS_MAKE ("{ GRAY8, RGB, BGR, RGBx, BGRx, xRGB, xBGR, RGBA, BGRA, ARGB, ABGR }") \
     ", views = (int) 1, interlace-mode = (string) progressive"
 
+/**
+ * @brief The supported video formats
+ */
+typedef enum
+{
+  DIRECT_VIDEO_FORMAT_UNKNOWN = 0,
+
+  /* Single Channel, Default: GRAY8 */
+  DIRECT_VIDEO_FORMAT_GRAY8 = 1,
+
+  /* 3 Channels, Default: RGB */
+  DIRECT_VIDEO_FORMAT_RGB = 2,
+  DIRECT_VIDEO_FORMAT_BGR = 3,
+
+  /* 4 Channels, Default: BGRx */
+  DIRECT_VIDEO_FORMAT_RGBx = 4,
+  DIRECT_VIDEO_FORMAT_BGRx = 5,
+  DIRECT_VIDEO_FORMAT_xRGB = 6,
+  DIRECT_VIDEO_FORMAT_xBGR = 7,
+  DIRECT_VIDEO_FORMAT_RGBA = 8,
+  DIRECT_VIDEO_FORMAT_BGRA = 9,
+  DIRECT_VIDEO_FORMAT_ARGB = 10,
+  DIRECT_VIDEO_FORMAT_ABGR = 11,
+} direct_video_formats;
+
+/**
+ * @brief Data structure for direct video options.
+ */
+typedef struct
+{
+  /* From option1 */
+  direct_video_formats format;
+} direct_video_ops;
+
+/**
+ * @brief List of the formats of direct video
+ */
+static const char *dv_formats[] = {
+  [DIRECT_VIDEO_FORMAT_UNKNOWN] = "UNKNOWN",
+  [DIRECT_VIDEO_FORMAT_GRAY8] = "GRAY8",
+  [DIRECT_VIDEO_FORMAT_RGB] = "RGB",
+  [DIRECT_VIDEO_FORMAT_BGR] = "BGR",
+  [DIRECT_VIDEO_FORMAT_RGBx] = "RGBx",
+  [DIRECT_VIDEO_FORMAT_BGRx] = "BGRx",
+  [DIRECT_VIDEO_FORMAT_xRGB] = "xRGB",
+  [DIRECT_VIDEO_FORMAT_xBGR] = "xBGR",
+  [DIRECT_VIDEO_FORMAT_RGBA] = "RGBA",
+  [DIRECT_VIDEO_FORMAT_BGRA] = "BGRA",
+  [DIRECT_VIDEO_FORMAT_ARGB] = "ARGB",
+  [DIRECT_VIDEO_FORMAT_ABGR] = "ABGR",
+  NULL,
+};
+
 /** @brief tensordec-plugin's GstTensorDecoderDef callback */
 static int
 dv_init (void **pdata)
 {
-  *pdata = NULL;                /* We have no internal data */
+  direct_video_ops *ddata;
+  ddata = *pdata = g_try_new0 (direct_video_ops, 1);
+  ddata->format = DIRECT_VIDEO_FORMAT_UNKNOWN;
```

Comment:
This line should be located after NULL check.

Reply:
nice catch :+1: